components/test/brave_player/playerEmbed_test.ts

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import {
+  EmbedURL,
+  getFrameSrc
+} from '../../brave_player/browser/resources/playerEmbed'
+
+describe('playerEmbedTest', () => {
+  describe('getFrameSrc', () => {
+    it(`returns empty string if pathname doesn't have video id or isn't for youtube`, () => {
+      expect(
+        getFrameSrc(new URL('chrome-untrusted://player-embed/youtube').pathname)
+      ).toBe('')
+      expect(
+        getFrameSrc(new URL('chrome-untrusted://player-embed').pathname)
+      ).toBe('')
+      expect(
+        getFrameSrc(new URL('chrome-untrusted://player-embed/vimeo').pathname)
+      ).toBe('')
+    })
+
+    it('returns YouTube embed URL with given video id', () => {
+      const videoId = 'dQw4w9WgXcQ'
```

Comment:
`brave://player/youtube/dQw4w9WgXcQ` doesn't actually work for me, says `Video unavailable` 
<img width="797" alt="image" src="https://github.com/brave/brave-core/assets/10810135/e23fe1ed-bbaf-4857-9c86-d623b1d7b742">
I assume that embedded playing is blocked for this video, which is fine. The issue is that navigation to `Watch on YouTube` is blocked because of COOP:
<img width="1416" alt="image" src="https://github.com/brave/brave-core/assets/10810135/a52f165b-a35b-4a6b-8ad4-da7047002b7e">

Reply:
@pes10k do you want the 'watch on youtube' button to work? if so i don't think we want to modify the COOP headers on youtube, so would have to add some workaround that involves injecting a content script that detects when the button is clicked and maybe using a Chromium API to actually open the new tab that goes to youtube (or navigate to youtube in the same tab).

Reply:
noting that 'watch on youtube' also appears on embeds usually if you hover over the 'youtube' icon at the bottom. haven't tested it specifically with brave player though.

Reply:
Discussed this with @diracdeltas in Slack. Adding `allow-popups-to-escape-sandbox` attribute to iframe's `sandbox` element fixed this issue. @sangwoo108 can you add it here https://github.com/brave/brave-core/blob/6a15caf07b87fe1804479f3dac02c1f3a65aebd0/components/brave_player/browser/resources/playerEmbed.tsx#L49?

Reply:
wait, please don't do that, at least not yet. A primary goal of this feature is to make these embeds indistinguishable from the plain YT suggested iframe element.

There shouldn't be any sandbox element.  I see this in the default code:

```<iframe width="560" height="315" src="<URL>" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>```

Or this if you use the 'privacy enhanced' verison of the embed

```<iframe width="560" height="315" src="<URL>" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>```

Again, its critical that we not make these embeds look different than other embeds on the web

Reply:
This was discussed on slack here: https://bravesoftware.slack.com/archives/CELA35G5S/p1698201001169889?thread_ts=1697639429.866849&cid=CELA35G5S

Reply:
@pes10k , I think the both examples are identical. Maybe privacy enhanced version is not pasted well?

Reply:
@stoletheminerals Thanks you, there was some videos unavailable ,and I was suspecting the videos have copyrighted music, but have yet to find a way to solve it. Do you think we can solve it without `sandbox` as per the @pes10k 's comment?

Reply:
@sangwoo108 yep, it works as expected without `sandbox`. thanks!